web_src/js/features/common-issue.js

```diff
@@ -2,14 +2,33 @@ import $ from 'jquery';
 import {updateIssuesMeta} from './repo-issue.js';
 
 export function initCommonIssue() {
-  $('.issue-checkbox').on('click', () => {
-    const numChecked = $('.issue-checkbox').children('input:checked').length;
-    if (numChecked > 0) {
+  $('.issue-checkbox,.issue-checkbox-all').on('click', (e) => {
+    const issuecheckbox = $('.issue-checkbox input');
+    if (e.currentTarget.className.includes('issue-checkbox-all')) {
+      if ($('.issue-checkbox-all input').prop('checked')) {
+        const selected = $('.issue-checkbox input:checked');
+        $('.issue-checkbox input:not(:checked)').prop('checked', 1);
+        selected.prop('checked', 0);
+      } else {
+        $('.issue-checkbox input:checked').prop('checked', 0);
+      }
+    }
+    if (e.shiftKey && window.config.checkboxfirst !== undefined) {
+      for (let i = window.config.checkboxfirst + 1, j = issuecheckbox.index($(e.currentTarget).find('input')); i < j; i++) {
+        issuecheckbox[i].checked = 1;
+      }
+      delete window.config.checkboxfirst;
+    } else {
+      window.config.checkboxfirst = issuecheckbox.index($(e.currentTarget).find('input'));
+    }
```

Comment:
`window.config` shouldn't be used to store data. This rather should be a variable declared within this function.